plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -1172,6 +1177,7 @@ def __init__(self):
         self.os_type = None
         self.os_disk_caching = None
         self.os_disk_size_gb = None
+        self.os_disk_encryption_set = None
```

Comment:
Please add required_if = [('os_disk_encryption_set', '*', ['managed_disk_type'])] and add "required_if=required_if" to line 1219